code/modules/mob/living/living_say.dm

```diff
@@ -315,16 +354,26 @@ GLOBAL_LIST_INIT(department_radio_keys, list(
 	return TRUE
 
 /mob/living/proc/can_speak_vocal(message) //Check AFTER handling of xeno and ling channels
+	var/mob/living/carbon/human/H = src
 	if(HAS_TRAIT(src, TRAIT_MUTE))
-		return FALSE
+		if(HAS_TRAIT(src, TRAIT_SIGN_LANG) && !H.mind.miming) //Makes sure mimes can't speak using sign language
+			return 1
+		else
+			return 0
 
 	if(is_muzzled())
-		return FALSE
+		if(HAS_TRAIT(src, TRAIT_SIGN_LANG) && !H.mind.miming)
+			return 1
+		else
+			return 0
 
 	if(!IsVocal())
-		return FALSE
+		if(HAS_TRAIT(src, TRAIT_SIGN_LANG) && !H.mind.miming)
+			return 1
+		else
+			return 0
 
-	return TRUE
+	return 1
```

Comment:
Not sure why you removed the boolean names?

Reply:
Apologies, I used a slightly out of date version of TG to code this, however I made sure that everything (hopefully) works correctly.
I need to fix the boolean names though, I'll get on that asap.